src/util/pg.py

```diff
@@ -472,6 +473,30 @@ def _column_info(cr, table, column):
     return cr.fetchone()
 
 
+_COLUMNS = {}
+if version_gte("10.0"):  # Columns were added in 9.0
+    _COLUMNS[("ir_ui_view", "arch_db")] = True
+    _COLUMNS[("ir_filters", "sort")] = True
+    _COLUMNS[("ir_model_fields", "relation_table")] = True
+    _COLUMNS[("ir_model_fields", "depends")] = True
+if version_gte("14.0"):  # Columns added in 13.0
+    _COLUMNS[("ir_module_module_dependency", "auto_install_required")] = False
+
+
+def _forced_cache(cache):
+    def decorator(f):
+        @functools.wraps(f)
+        def wrapper(cr, *args, **kwargs):
+            if not kwargs and tuple(args) in cache:
+                return cache[tuple(args)]
+            return f(cr, *args, **kwargs)
+
+        return wrapper
+
+    return decorator
+
+
+@_forced_cache(_COLUMNS)
```

Comment:
I don't think the decorator is needed. It can be inline into the function directly.
```python
def column_exists(cr, table, column):
    if (table, column) in _COLUMNS:
        return _COLUMNS[(table, column)]
    return _column_info(cr, table, column) is not None
```

Reply:
Well, originally I was using it in `get_value_or_en_translation`. It proved not as useful. I'm still working on this, I will simplify as much as possible before the final version. There are still some other columns being checked more than 70 times in the local test I'm doing.